tests/test_job.py

```diff
@@ -55,7 +55,7 @@ def test_get_source_project_when_is_fork(self):
     def test_get_mr_ci_status(self, version, use_merge_request_pipelines):
         with patch('marge.job.Pipeline', autospec=True) as pipeline_class:
             pipeline_success = [
-                Mock(spec=pipeline_class, sha='abc', status='success'),
+                Mock(sha='abc', status='success'),
```

Comment:
This test failed on Python 3.11. I guess removing the `spec` part technically is making it worse, but it seems to work :shrug: And I couldn't find any other way of getting it to work.

```
FAILED tests/test_job.py::TestJob::test_get_mr_ci_status[9.4.0-ee-False] - unittest.mock.InvalidSpecError: Cannot spec a Mock object. [object=<MagicMock name='Pipeline' spec='Pipeline' id='140056590036240'>]
FAILED tests/test_job.py::TestJob::test_get_mr_ci_status[10.5.0-ee-True] - unittest.mock.InvalidSpecError: Cannot spec a Mock object. [object=<MagicMock name='Pipeline' spec='Pipeline' id='140056592312080'>]
```


```
__________________________________________________________________________________________________________ TestJob.test_get_mr_ci_status[10.5.0-ee-True] __________________________________________________________________________________________________________

self = <tests.test_job.TestJob object at 0x7f617a1fca10>, version = '10.5.0-ee', use_merge_request_pipelines = True

    @pytest.mark.parametrize(
        'version,use_merge_request_pipelines',
        [('9.4.0-ee', False), ('10.5.0-ee', True)],
    )
    def test_get_mr_ci_status(self, version, use_merge_request_pipelines):
        with patch('marge.job.Pipeline', autospec=True) as pipeline_class:
            pipeline_success = [
>               Mock(spec=pipeline_class, sha='abc', status='success'),
            ]

tests/test_job.py:58: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
../../../.pyenv/versions/3.11.1/lib/python3.11/unittest/mock.py:1100: in __init__
    _safe_super(CallableMixin, self).__init__(
../../../.pyenv/versions/3.11.1/lib/python3.11/unittest/mock.py:451: in __init__
    self._mock_add_spec(spec, spec_set, _spec_as_instance, _eat_self)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <[AttributeError('_mock_methods') raised in repr()] Mock object at 0x7f61776ee5d0>, spec = <MagicMock name='Pipeline' spec='Pipeline' id='140056592312080'>, spec_set = None, _spec_as_instance = False, _eat_self = False

    def _mock_add_spec(self, spec, spec_set, _spec_as_instance=False,
                       _eat_self=False):
        if _is_instance_mock(spec):
>           raise InvalidSpecError(f'Cannot spec a Mock object. [object={spec!r}]')
E           unittest.mock.InvalidSpecError: Cannot spec a Mock object. [object=<MagicMock name='Pipeline' spec='Pipeline' id='140056592312080'>]

../../../.pyenv/versions/3.11.1/lib/python3.11/unittest/mock.py:502: InvalidSpecError
```